assets/questions/bstClosestValue.js

```diff
@@ -0,0 +1,216 @@
+  title: 'Find Closest Value in Binary Search Tree',
+  starterCode: {
+    python:
+`def findClosestValue(input):
+    root = input[0]
+    target = input[1]
+
+    #TODO : write your code after here
+    return ""
+`
+  },
+  auxiliaryCode: {
+    python:
+`class AuxiliaryCode(object):
+        @classmethod
+        def deserialize(cls, data):
+            class TreeNode(object):
+                def __init__(self, x):
+                    self.val = x
+                    self.left = None
+                    self.right = None
+
+
+                def __str__(self):
+                    return self.val
+
+            def constructTree():
+                val = next(vals)
+                if val == '#':
+                    return None
+                node = TreeNode(float(val))
+                node.left = constructTree()
+                node.right = constructTree()
+                return node
+
+            inputs = data.split('|')
+            vals = iter(inputs[0].split())
+            return constructTree(), float(inputs[1])
+
+        @classmethod
+        def failToUseAbsoluteDiff(cls, input):
+            root = input[0]
+            target = input[1]
+
+            kid = root.left if target < root.val else root.right
+            if not kid:
+                return root.val
+            closest = cls.failToUseAbsoluteDiff([kid,target])
+            return root.val if (root.val - target) < (closest - target) else closest
+
+        @classmethod
+        def failToCompareAll(cls, input):
+            root = input[0]
+            target = input[1]
+
+            kid = root.left if target < root.val else root.right
+            if not kid:
+                return root.val
+            return cls.failToCompareAll([kid,target])
+`
+  },
+  tasks: [{
+    instructions: [
+      {
+        content:
+        [
+          'In this question, you need to find the closest value in a' +
+          ' non-empty binary search tree ',
+          'given the target value. The target value is given in float' +
+          ' format. It\'s guaranteed ',
+          'that there\'s only one closest value. You will have the root' +
+          ' TreeNode. Each TreeNode ',
+          'has attributes val, left and right.'
+        ].join(''),
+        type: 'text'
+      },
+      {
+        content:
+        [
+          'For instance, given the following tree and the target value 1.1,' +
+          ' you should return 1.',
+          '  2',
+          ' / \\',
+          '1   3'
+        ].join('\n'),
+        type: 'code'
+      },
+      {
+        content:
+        [
+          'The tree is represented as a preorder traversal sequence of node' +
+          ' values in which \'#\'',
+          'represents an empty node. For example: the input \'2 1 # # #\'' +
+          ' represents the serialized',
+          'format of a binary tree using preorder order traversal, where' +
+          ' root value is 2 and only has a left child with value 1.'
```

Comment:
The call for action is in the first paragraph, which is "In this question, you need to find the closest value in a non-empty binary search tree, given the target value."